### ISSUE TITLE
nixFlakes doesn't handle ninja's progress indication properly

### ISSUE THREAD

**Issue by zseri** [NixOS/nix#4631]
**Describe the bug**

nix (with flakes enabled, and with the system config in a flake), it doesn't handle output from build processes which call `ninja` properly, as it doesn't show the process of the ninja process (e.g. what it currently compiles).

(assumption) as ninja terminates most lines with `\r`, and no lines for ninja building stuff is output, until either (1) it is done, or (2) a warning from C compiler, etc. is printed when hit.

**Steps To Reproduce**

1. Enable `nixFlakes`
2. Add a package to `systemPackages` which uses ninja
3. `nixos-rebuild --show-trace -L -v switch`
4. it doesn't show the progress of ninja properly, as described above

**Expected behavior**

when used with `-L -v`, it should output all lines of the build process, which end in `\r`, with `\n` instead, e.g. replace `\r` with `\n` in the underlying build output. when used with the progress bar only (e.g. without `-L -v`), it should integrate ninja's progress bar, by printing the output of ninja after it's own progress bar, and reprint it's progress bar + ninja progress, when ninja prints the next progress indication line terminated with `\r`.

**`nix-env --version` output**
nix-env (Nix) 2.4pre20201201_5a6ddb3

**Additional context**

currently none.

**Comment by RossComputerGuy** [comment]
I don't think this is a Flakes specific issue, more of a problem of how Nix handles the log.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
